Sources/TuistKit/Commands/TestCommand.swift

```diff
@@ -130,20 +130,80 @@ public struct TestCommand: AsyncParsableCommand, HasTrackableParameters {
     )
     var generateOnly: Bool = false
 
+    @Argument(
+        parsing: .postTerminator,
+        help: "xcodebuild arguments that will be passthrough"
+    )
+    var passthroughXcodeBuildArguments: [String] = []
+
     public func validate() throws {
         try TestService().validateParameters(
             testTargets: testTargets,
             skipTestTargets: skipTestTargets
         )
     }
 
+    private var notAllowedPassthroughXcodeBuildArguments = [
+        "-scheme",
+        "-workspace",
+        "-project",
+        "-testPlan",
+        "-skip-test-configuration",
+        "-only-test-configuration",
+        "-only-testing",
+        "-skip-testing",
+    ]
+
     public func run() async throws {
-        let absolutePath: AbsolutePath
+        // Check if passthrough arguments are already handled by tuist
+        try notAllowedPassthroughXcodeBuildArguments.forEach {
+            if passthroughXcodeBuildArguments.contains($0) {
+                throw XcodeBuildPassthroughArgumentError.alreadyHandled($0)
+            }
+        }
+
+        // Suggest the user to use passthrough arguments if already supported by xcodebuild
+        if platform != nil || os != nil || device != nil || rosetta {
```

Comment:
There's a bit of duplication between `BuildCommand` and `TestCommand`, but considering this logic will eventually go away, and that we don't plan to change it, I think it's an acceptable duplication.

Reply:
Yes, that was my train of thought.